### PR TITLE
settings: fix settings_line_val_get_len while base64 encoding is enabled

### LINKED ISSUE #12122: system.settings.nffs fails on NRF5 boards
**Describe the bug**
system.settings.nffs at tests/subsys/settings/nffs fails on NRF5 boards.

Boards which I tested are: nrf52840_pca10056 and nrf52_pca10040. The tests fails on both the boards.

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=nrf52_pca10040 
3. make

**Screenshots or console output**
```
***** Booting Zephyr OS zephyr-v1.13.0-2782-g07eb5d1 (delayed boot 1000ms) *****
Running test suite test_config_fcb
===================================================================
starting test - config_empty_lookups
PASS - config_empty_lookups
===================================================================
starting test - test_config_insert
PASS - test_config_insert
===================================================================
starting test - test_config_getset_unknown
PASS - test_config_getset_unknown
===================================================================
starting test - test_config_getset_int
PASS - test_config_getset_int
===================================================================
starting test - test_config_getset_int64
PASS - test_config_getset_int64
===================================================================
starting test - test_config_commit
PASS - test_config_commit
===================================================================
starting test - config_setup_nffs
PASS - config_setup_nffs
===================================================================
starting test - test_config_empty_file
PASS - test_config_empty_file
===================================================================
starting test - test_config_small_file

    Assertion failed at /home/kspoorth/work/latest_zephyr/tests/subsys/settings/nffs/src/settings_test_nffs.c:75: c1_handle_set: (val_len == 1 is false)
bad set-value size
FAIL - test_config_small_file
===================================================================
starting test - test_config_multiple_in_file

    Assertion failed at /home/kspoorth/work/latest_zephyr/tests/subsys/settings/nffs/src/settings_test_nffs.c:75: c1_handle_set: (val_len == 1 is false)
bad set-value size
FAIL - test_config_multiple_in_file
===================================================================
starting test - test_config_save_in_file
PASS - test_config_save_in_file
===================================================================
starting test - test_config_save_one_file

    Assertion failed at /home/kspoorth/work/latest_zephyr/tests/subsys/settings/nffs/src/settings_test_nffs.c:75: c1_handle_set: (val_len == 1 is false)
bad set-value size
FAIL - test_config_save_one_file
===================================================================
starting test - test_config_compress_file

    Assertion failed at /home/kspoorth/work/latest_zephyr/tests/subsys/settings/nffs/src/settings_test_nffs.c:75: c1_handle_set: (val_len == 1 is false)
bad set-value size
FAIL - test_config_compress_file
===================================================================
Test suite test_config_fcb failed.
===================================================================
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.9.5
 - Commit SHA or Version used: 07eb5d1eaa584aac7fb20ab28abf171f2bec62e9

### PR DESCRIPTION
**Pull request by nvlsianpu**
While base64 encoding was enabled settings_val_get_len_cb() returned
encoded length instead of decoded value length.

This patches introduce procedure which calculates the value length
properly. For that additional read of the end of the encoded value is
required for checking how many form last 3 value bytes wear encoded.

fixes #12122

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: subsys/settings/src/settings_line.c [hunk 1]
```diff
@@ -438,7 +438,38 @@ size_t settings_line_val_get_len(off_t val_off, void *read_cb_ctx)
...
+	size_t len_base64 = len - val_off;
+
+	/* don't care about lack of alignmet to 4 B */
+	/* entire value redout call will return error anyway */
+	if (len_base64 >= 4) {
+		/* read last 2 B of base64 */
+		rc = settings_line_raw_read(len - 2, raw, 2, &len, read_cb_ctx);
+		if (rc || len != 2) {
+			/* very unexpected error */
+			__ASSERT(rc == 0, "Failed to read the storage.\n");
```
Comment (nashif) [inline subsys/settings/src/settings_line.c]: need to include assert.h for this.

**Review by nashif — CHANGES_REQUESTED** [review]

File: subsys/settings/src/settings_line.c (hunk 1 above)
Comment (nvlsianpu) [reply subsys/settings/src/settings_line.c]: fixed

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
LGTM

**Comment by nvlsianpu** [comment]
@nashif Can you approve and merge this bug-fix?

**Review by nashif — APPROVED** [review]

**Pull request merged**